src/OgAccess.php

```diff
@@ -112,18 +111,24 @@ public static function userAccess(EntityInterface $group, $operation, AccountInt
     // To reduce the number of SQL queries, we cache the user's permissions
     // in a static variable.
     if (!$pre_alter_cache) {
-      $permissions = array();
-
-      foreach (Og::getUserMemberships($user) as $membership) {
+      $permissions = [];
+      $user_is_group_admin = FALSE;
 
+      if ($membership = Og::getUserMembership($user, $group)) {
         foreach ($membership->getRoles() as $role) {
-
           /** @var $role RoleInterface */
-          $permissions = array_unique(array_merge($permissions, $role->getPermissions()));
+          $permissions = array_merge($permissions, $role->getPermissions());
+
+          // Set an is_admin flag.
+          if ($role->isAdmin()) {
```

Comment:
I think that if a role isAdmin, we don't need to continue iterating. We can just add all the permissions